packages/react-server/src/ReactFlightReplyServer.js

```diff
@@ -397,6 +397,22 @@ function parseModelString(
           key,
         );
       }
+      case 'I': {
+        // $Infinity
+        return Infinity;
+      }
+      case '-': {
+        // $-0 or $-Infinity
+        if (value[2] === '0') {
```

Comment:
Because of interning and that you don’t have to do a type check, so there’s nothing to recompile during a deopt, it might be better to just compare to the full string.

The reason not to do it for the outer check is that the common case needs to extract a partial anyway so we’re just using the same case.

Reply:
ok https://github.com/facebook/react/pull/26629

Reply:
for the outer switch – I was thinking about it last night and I wonder if switching on charCodeAt is likely faster?